sklearnex/ensemble/forest.py

```diff
@@ -16,3 +16,1146 @@
+                             "%r" % self.max_bins)
+        if isinstance(self.min_bin_size, numbers.Integral):
+            if not 1 <= self.min_bin_size:
+                raise ValueError("min_bin_size must be at least 1, got %s"
+                                 % self.min_bin_size)
+        else:
+            raise ValueError("min_bin_size must be integral number but was "
+                             "%r" % self.min_bin_size)
+
+    def check_sample_weight(self, sample_weight, X, dtype=None):
+        n_samples = _num_samples(X)
+
+        if dtype is not None and dtype not in [np.float32, np.float64]:
+            dtype = np.float64
+
+        if sample_weight is None:
+            sample_weight = np.ones(n_samples, dtype=dtype)
+        elif isinstance(sample_weight, numbers.Number):
+            sample_weight = np.full(n_samples, sample_weight, dtype=dtype)
+        else:
+            if dtype is None:
+                dtype = [np.float64, np.float32]
+            sample_weight = check_array(
+                sample_weight,
+                accept_sparse=False,
+                ensure_2d=False,
+                dtype=dtype,
+                order="C")
+            if sample_weight.ndim != 1:
+                raise ValueError("Sample weights must be 1D array or scalar")
+
+            if sample_weight.shape != (n_samples,):
+                raise ValueError("sample_weight.shape == {}, expected {}!"
+                                 .format(sample_weight.shape, (n_samples,)))
+        return sample_weight
+
+
+class ExtraTreesClassifier(sklearn_ExtraTreesClassifier, BaseForest):
+    __doc__ = sklearn_ExtraTreesClassifier.__doc__
+
+    if sklearn_check_version('1.2'):
+        _parameter_constraints: dict = {
+            **sklearn_ExtraTreesClassifier._parameter_constraints,
+            "max_bins": [Interval(numbers.Integral, 2, None, closed="left")],
+            "min_bin_size": [Interval(numbers.Integral, 1, None, closed="left")],
+            "splitter_mode": [StrOptions({"best", "random"})]
+        }
+
+    if sklearn_check_version('1.0'):
+        def __init__(
+                self,
+                n_estimators=100,
+                criterion="gini",
+                max_depth=None,
+                min_samples_split=2,
+                min_samples_leaf=1,
+                min_weight_fraction_leaf=0.,
+                max_features='sqrt' if sklearn_check_version('1.1') else 'auto',
+                max_leaf_nodes=None,
+                min_impurity_decrease=0.,
+                bootstrap=False,
+                oob_score=False,
+                n_jobs=None,
+                random_state=None,
+                verbose=0,
+                warm_start=False,
+                class_weight=None,
+                ccp_alpha=0.0,
+                max_samples=None,
+                max_bins=256,
+                min_bin_size=1,
+                splitter_mode='random'):
```

Comment:
There is no `splitter_mode` param for `ExtraTreesClassifier`

Reply:
I don't understand. The splitter_mode is not passed to stock sklearn

Reply:
Stock sklearn doesn't have such param: https://github.com/scikit-learn/scikit-learn/blob/364c77e047ca08a95862becf40a04fe9d4cd2c98/sklearn/ensemble/_forest.py#L2072
no need to extend API on sklearnex.

Reply:
https://github.com/intel/scikit-learn-intelex/blob/master/sklearnex/preview/ensemble/forest.py#L223
https://github.com/intel/scikit-learn-intelex/blob/master/sklearnex/preview/ensemble/forest.py#L776
Should we then change RandomForestClassifier and RandomForestRegressor as well?

Reply:
No, let's leave it for RF. SInce ve don't have common ForestCLassifier with these param. And we using this for param for onedal estimators. For extratrees no need.

Reply:
Unfortunately this solution is implementation inconsistent. I guess I don't understand your justification to maintain for RF and not for ExtraTrees.

Reply:
Due to lack of time, I have removed splitter_mode from the public API. This is a point we need further discussion on RF side when the refactor occurs soon. This will be set on a level farther down, and the splitter_mode access in preview will not be used further on.

Reply:
It doesn't make any sense with new proposal. If are going to have `ForestClassifier/Regressor` (that allows using `splitter_mode`) so there is not need to support in `RandomForestCLassifier/Regressor` API. So my previous comments are outdated now. Please ignore them.